src/sparsezoo/refactor/directory.py

```diff
@@ -0,0 +1,194 @@
+                "were not successfully validated."
+            )
+
+        return all(validations.values())
+
+    def download(self, destination_path: str, overwrite: bool = True):
+        """
+        Download the contents of the file from the url.
+
+        :param destination_path: the local file path to save the downloaded file to
+        :param overwrite: True to overwrite any previous files if they exist,
+            False to not overwrite and raise an error if a file exists
+        """
+        # Directory can represent a tar file.
+        if self._is_tar():
+            download_file(
+                url_path=self.url,
+                dest_path=os.path.join(destination_path, self.name),
+                overwrite=overwrite,
+            )
+            self.path = os.path.join(destination_path, self.name)
+
+        # Directory can represent a folder.
+        else:
+            for file in self.files:
+                if isinstance(file, Directory):
+                    raise NotImplementedError(
+                        "Method download() does not support nested directories yet."
+                    )
+                else:
+                    file.download(destination_path=destination_path)
+
+    @classmethod
+    def gzip(
+        cls, directory: "Directory", archive_directory: Optional[str] = None
+    ) -> "Directory":
+        """
+        Factory method that creates a tar archive Directory
+        from the `directory` (folder Directory).
+        The tar archive file would be saved in the parent
+        directory of the `directory`, unless `extract_directory` argument is specified.
+
+        :param directory: the folder directory (Directory class object)
+        :param archive_directory: the local path to
+            save new tar archive Directory to (default = None)
+        :return: tar archive Directory
+        """
+        if directory._is_tar():
+            raise ValueError(
+                "Attempting to create a tar archive "
+                "Directory from a tar archive Directory."
+            )
+        if archive_directory is not None:
+            parent_path = archive_directory
+        else:
+            if directory.path is None:
+                raise ValueError(
+                    "Attempting to zip the folder Directory object files using "
+                    "`path` attribute, but `self.path` is None. "
+                    "Class object requires pointer to parent "
+                    "folder directory to know where to save the tar archive file."
+                )
+            parent_path = pathlib.PurePath(directory.path).parent
+
+        tar_file_name = directory.name + ".tar.gz"
+        tar_file_path = os.path.join(parent_path, tar_file_name)
+        with tarfile.open(tar_file_path, "w") as tar:
+            for file in directory.files:
+                tar.add(file.path)
+        return Directory(name=tar_file_name, path=tar_file_path)
+
+    @classmethod
+    def unzip(
```

Comment:
why make this a classmethod if we are taking a `Directory` as input? any reason to not do `self.unzip`?

Reply:
Fixed.